shared/common-adapters/name-with-icon/container.js

```diff
@@ -23,27 +23,29 @@ const mapDispatchToProps = dispatch => ({
 })
 
 const mergeProps = (stateProps, dispatchProps, ownProps: ConnectedNameWithIconProps) => {
-  const {onClick, ...props} = ownProps
+  const {onClick, username, teamname, ...props} = ownProps
 
   let functionOnClick
   let clickType
   // Since there's no tracker on mobile, we can't open it. Fallback to profile.
   if (!isMobile && onClick === 'tracker') {
-    if (ownProps.username) {
-      functionOnClick = dispatchProps.onOpenTracker(ownProps.username)
+    if (username) {
+      functionOnClick = () => dispatchProps.onOpenTracker(username)
```

Comment:
the weird flow error here was probably because you had
```
() => dispatchProps.onOpenTracker(ownProps.username)
```
which closes over `ownProps`, which may get mutated before the closure is called (and so the type of `ownProps.username` can't be refined to `string` from `?string`)